php/utils.php

```diff
@@ -1505,3 +1504,72 @@ function format_table( $items ) {
 	}
 	\EE::log( $delem );
 }
+
+/**
+ * Get the site-name from the path from where ee is running if it is a valid site path.
+ *
+ * @return bool|String Name of the site or false in failure.
+ */
+function get_site_name() {
+	$sites = EE::db()::select( array( 'sitename' ) );
+
+	if ( $sites ) {
+		$cwd          = getcwd();
+		$name_in_path = explode( '/', $cwd );
+		$site_name    = array_intersect( array_flatten( $sites ), $name_in_path );
+
+		if ( 1 === count( $site_name ) ) {
+			$path = EE::db()::select( array( 'site_path' ), array( 'sitename' => $site_name[0] ) );
+			if ( $path ) {
+				$site_path = $path[0]['site_path'];
+				if ( $site_path === substr( $cwd, 0, strlen( $site_path ) ) ) {
+					return $site_name[0];
+				}
+			}
+		}
+	}
+
+	return false;
```

Comment:
Now this function can return (or trigger) two types of error - 
1. site not found
2. duplicate site found.

In order to handle this, instead of returning integers, we can throw exceptions from this functions and catch it in parent function.

Reply:
Comment isn't relevant as if condition is being removed.